src/test/javascript/spec/component/orion/orion-programming-exercise.component.spec.ts

```diff
@@ -34,34 +38,36 @@ describe('OrionProgrammingExerciseComponent', () => {
     });
 
     it('ngOnInit should subscribe to state', () => {
-        const orionStateStub = jest.spyOn(orionConnectorService, 'state');
         const orionState = { opened: 40, building: false, cloning: false } as any;
-        orionStateStub.mockReturnValue(new BehaviorSubject(orionState));
+        const orionStateStub = jest.spyOn(orionConnectorService, 'state').mockReturnValue(new BehaviorSubject(orionState));
 
         comp.ngOnInit();
 
         expect(orionStateStub).toHaveBeenCalledTimes(1);
         expect(orionStateStub).toHaveBeenCalledWith();
         expect(comp.orionState).toEqual(orionState);
     });
+
     it('editInIde should call connector', () => {
         const editExerciseSpy = jest.spyOn(orionConnectorService, 'editExercise');
-        jest.spyOn(TestBed.inject(ProgrammingExerciseService), 'find').mockReturnValue(new BehaviorSubject({ body: programmingExercise } as any));
+        jest.spyOn(programmingExerciseService, 'find').mockReturnValue(new BehaviorSubject({ body: programmingExercise } as any));
 
         comp.editInIDE(programmingExercise);
 
         expect(editExerciseSpy).toHaveBeenCalledTimes(1);
         expect(editExerciseSpy).toHaveBeenCalledWith(programmingExercise);
     });
+
     it('openOrionEditor should navigate to orion editor', () => {
-        const navigateSpy = jest.spyOn(TestBed.inject(Router), 'navigate');
+        const navigateSpy = jest.spyOn(router, 'navigate');
         comp.openOrionEditor({ ...programmingExercise, templateParticipation: { id: 1234 } });
 
         expect(navigateSpy).toHaveBeenCalledTimes(1);
         expect(navigateSpy).toHaveBeenCalledWith(['code-editor', 'ide', 456, 'admin', 1234]);
     });
-    it('openOrionEditor with error', () => {
-        const navigateStub = jest.spyOn(TestBed.inject(Router), 'navigate').mockImplementation(() => {
+
+    it('openOrionEditor should handle error', () => {
+        const navigateStub = jest.spyOn(router, 'navigate').mockImplementation(() => {
             throw 'test error';
```

Comment:
I'd suggest using the `throwError` function you used somewhere else already too.

Reply:
This does not work; On the other occurence we mocked an Observable that throws an error, here we just throw an error